java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -53,6 +59,25 @@
 
   @Override
   protected void onConstructorFound(NewClassTree newClassTree) {
+    if(isBigIntegerPotentiallyBiggerThanLong(newClassTree)) {
+      return;
+    }
     reportIssue(newClassTree.identifier(), "Remove this \"" + newClassTree.symbolType().name() + "\" constructor");
   }
+
+  private static boolean isBigIntegerPotentiallyBiggerThanLong(NewClassTree newClassTree) {
+    if (!newClassTree.symbolType().is("java.math.BigInteger")) {
+      return false;
+    }
+    ExpressionTree argument = newClassTree.arguments().get(0);
+    if (!argument.is(Tree.Kind.STRING_LITERAL)) {
+      return true;
+    }
+    try {
+      BigInteger value = new BigInteger(LiteralUtils.trimQuotes(((LiteralTree)argument).value()));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this "BigInteger" constructor ([squid:S2129](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2129))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=2012&issues=AWNPBJYoqAyH-7GyoNPi&open=AWNPBJYoqAyH-7GyoNPi)

[ISSUE_KEY]: AWNPBJYoqAyH-7GyoNPi